Improve download files page layout

nlhkabu opened the pull request:
Closes #18732 and #18796

nlhkabu committed:
Only show wheel filters if there is more than one wheel

nlhkabu committed:
Update display of search and filters

nlhkabu committed:
Display files in table

nlhkabu commented:
WIP:

<img width="1040" height="792" alt="Screenshot From 2025-09-27 07-44-27" src="https://github.com/user-attachments/assets/40f17cbc-807a-4c16-8081-eeec10dc049b" />

TODO:

- [ ] Update select to show values in "pretty" format - e.g. `CP26` becomes `CPython 2.6`
- [ ] Update table to show values in "pretty" format - e.g. `CP26` becomes `CPython 2.6`
- [ ] Show "No files match your selection" message when user filters by a combination which returns no results
- [ ] Limit the filters based on what the user has already selected
- [ ] Provide quick option to reset the filters (?)

nlhkabu commented:
Question how should we format the ABI?

Example options for the ABI `cp36m`:

<img width="1041" height="196" alt="image" src="https://github.com/user-attachments/assets/77c2bdd1-7fe8-4f45-9d8b-33dd03176f1a" />

OR

<img width="1041" height="196" alt="image" src="https://github.com/user-attachments/assets/7ecf8f3c-253f-42a0-b4fa-dcd73eaf2465" />

merwok commented:
IMO it should not be just `m (pymalloc)`, but needs to include all the parts: `m` and other modifiers don’t make sense without the first part `cp36`.

Unless it’s guaranteed that an abi tag `cp36m` _always_ has an interpreter tag of `cp36`, but I am not sure about that.

nlhkabu commented:
Ok, thanks for your feedback @merwok, and for your comments on the associated issue.
I've added "Update URL based on filters" to the TODO list

re: ABIs, I've added a couple more options above - based on your feedback, I think the third option is best: `CPython 3.6m (pymalloc)`

nlhkabu committed:
Show filters if more than 3 files are present

nlhkabu committed:
Adjust table columns to create more room for longer tag names

merwok commented:
I suggested 3 as threshold: it’s not useful to filter a list of 1 item, or 2 items.
(One could argue for a minimum of 5 or 6 as well!)

nlhkabu commented:
Yep, currently only showing the filters when there are more than 3 wheels, but happy to adjust that value based on feedback :)

merwok commented:
I think that last bug can be solved here: https://github.com/pypi/warehouse/blob/main/warehouse/static/js/warehouse/controllers/filter_list_controller.js#L199-L208

cofiem commented:
My work in progress is here: https://github.com/pypi/warehouse/compare/main...cofiem:warehouse:feature/improve-download-files-page?expand=1
